atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionTestSetupV2.java

```diff
@@ -0,0 +1,295 @@
+import com.palantir.atlasdb.internalschema.InternalSchemaMetadata;
+import com.palantir.atlasdb.internalschema.TransactionSchemaManager;
+import com.palantir.atlasdb.internalschema.persistence.CoordinationServices;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.api.watch.LockWatchManagerInternal;
+import com.palantir.atlasdb.keyvalue.impl.Cells;
+import com.palantir.atlasdb.keyvalue.impl.KvsManager;
+import com.palantir.atlasdb.keyvalue.impl.TransactionManagerManager;
+import com.palantir.atlasdb.persistent.api.PersistentStore;
+import com.palantir.atlasdb.persistent.rocksdb.RocksDbPersistentStore;
+import com.palantir.atlasdb.protos.generated.TableMetadataPersistence.SweepStrategy;
+import com.palantir.atlasdb.sweep.queue.MultiTableSweepQueueWriter;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.atlasdb.transaction.api.TransactionManager;
+import com.palantir.atlasdb.transaction.knowledge.TransactionKnowledgeComponents;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.atlasdb.util.MetricsManagers;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.LockServerOptions;
+import com.palantir.lock.LockService;
+import com.palantir.lock.impl.LockServiceImpl;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.test.utils.SubdirectoryCreator;
+import com.palantir.timelock.paxos.InMemoryTimelockExtension;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+import com.palantir.util.Pair;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+/**
+ * Expectations for semantics of tests that are run as part of this class (not that the author thinks these
+ * assumptions are ideal, but this is documentation of the current state):
+ * - The timestamp and lock services are disjoint across individual tests.
+ * - The key value services may not be disjoint across individual tests.
+ */
+public abstract class TransactionTestSetupV2 {
+    @TempDir
+    public static File persistentStorageFolder;
+
+    private static PersistentStore persistentStore;
+
+    @BeforeAll
+    public static void storageSetUp() throws RocksDBException {
+        File storageDirectory =
```

Comment:
Exact replication of `TransactionTestSetupV2`, except
- `@TempDir` instead of `new TemporaryFolder()`
- `BeforeAll` instead of `BeforeClass`
- `AfterAll` instead of `AfterClass`
-  `InMemoryTimelockExtension` instead of `InMemoryTimeLockRule`
- `BeforeEach` instead of `Before`